esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -47,7 +47,6 @@
 #include "workunit.hpp"
 #include "cumulativetimer.hpp"
 #include <memory>
-#include "esdl_def_helper.hpp"
```

Comment:
You don't have to move this include to httpbinding.hpp.  Or, I don't understand why you've done that.  You also don't need any of the CMakeLists.txt changes.

I cloned your branch and was able to build without any of those changes.  They really complicate this PR.

Reply:
getServiceSchema was converted to a method instead of a separate function. That way it has access to other httpbinding methods as well as service implementations. For example, the service XML filename is retrieved from a httpbinding method implemented by the service (from generated code). It would need to be passed in otherwise. Other implementations of httpbinding automatically get the new method as well.

Looking forward to optimizations, items like the compiled transforms and cached schemas seemed better to me to store in the binding object instead of separately. Currently the method creates a new set of ESDL helpers on each call. As a method, it can store those in the binding object and only create them once. Knowing that I would probably add those objects to the httpbinding class, I added the header inclusion to the hpp file. Since the httpbinding class is used in so many places across the repo, I had to modify all of the CMakeLists files.

If the reasons above are not sufficient, I will return it to a separate function.

Reply:
@kenrowland

Reply:
edited: meant getServiceSchema is fine as a member function.